spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/JdbcChannelMessageStore.java

```diff
@@ -448,23 +462,25 @@ public MessageGroup getMessageGroup(Object groupId) {
 	@ManagedAttribute
 	public int getMessageGroupCount() {
 		return this.jdbcTemplate.queryForObject(
-				getQuery("SELECT COUNT(DISTINCT GROUP_KEY) from %PREFIX%CHANNEL_MESSAGE where REGION = ?"),
+				getQuery(Query.COUNT_GROUPS,
+						() -> "SELECT COUNT(DISTINCT GROUP_KEY) from %PREFIX%CHANNEL_MESSAGE where REGION = ?"),
```

Comment:
This supplier is pointless as well: it doesn't do any calculation and just returns a constant.

Reply:
Simpler than adding a second cache method just for this one case.